File: src/model_v4/entities/SBS_FILIALE.ts
import { Column, Entity, OneToMany, PrimaryColumn } from "typeorm";
import { SBS_OE } from "./SBS_OE";
import { SBS_SEGMENT } from "./SBS_SEGMENT";


@Entity("SBS_FILIALE",{schema:"sbsdb"})
export class SBS_FILIALE {

    @PrimaryColumn("bigint",{ 
        nullable:false,
        primary:true,
        name:"FILIALE_INDEX"
        })
    FILIALE_INDEX:string;
        

    @Column("varchar",{ 
        nullable:true,
        length:50,
        default: () => "'NULL'",
        name:"HAUSNR"
        })
    HAUSNR:string | null;
        

    @Column("varchar",{ 
        nullable:true,
        length:50,
        default: () => "'NULL'",
        name:"ORT"
        })
    ORT:string | null;
        

    @Column("varchar",{ 
        nullable:true,
        length:50,
        default: () => "'NULL'",
        name:"PLZ"
        })
    PLZ:string | null;
        

    @Column("varchar",{ 
        nullable:true,
        length:50,
        default: () => "'NULL'",
        name:"STRASSE"
        })
    STRASSE:string | null;
        

   
    @OneToMany(type=>SBS_OE, SBS_OE=>SBS_OE.fILIALE_INDEX,{ onDelete: 'RESTRICT' ,onUpdate: 'RESTRICT' })
    sBS_OEs:SBS_OE[];
    

   
    @OneToMany(type=>SBS_SEGMENT, SBS_SEGMENT=>SBS_SEGMENT.fILIALE_INDEX,{ onDelete: 'RESTRICT' ,onUpdate: 'RESTRICT' })
    sBS_SEGMENTs:SBS_SEGMENT[];
    
}

File: SbsdbServer/sbsdbv4/model/SbsFiliale.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace hb.SbsdbServer.sbsdbv4.model
{
    [Table("SBS_FILIALE")]
    public partial class SbsFiliale
    {
        public SbsFiliale()
        {
            SbsOe = new HashSet<SbsOe>();
            //SbsSegment = new HashSet<SbsSegment>();
        }

        [Column("FILIALE_INDEX", TypeName = "bigint(20)")]
        public long FilialeIndex { get; set; }
        [Column("HAUSNR", TypeName = "varchar(50)")]
        public string Hausnr { get; set; }
        [Column("ORT", TypeName = "varchar(50)")]
        public string Ort { get; set; }
        [Column("PLZ", TypeName = "varchar(50)")]
        public string Plz { get; set; }
        [Column("STRASSE", TypeName = "varchar(50)")]
        public string Strasse { get; set; }

        [InverseProperty("FilialeIndexNavigation")]
        public virtual ICollection<SbsOe> SbsOe { get; set; }
        //[InverseProperty("FilialeIndexNavigation")]
        //public virtual ICollection<SbsSegment> SbsSegment { get; set; }
    }
}
